packages/langium/src/workspace/documents.ts

```diff
@@ -339,16 +383,34 @@ export class DefaultLangiumDocuments implements LangiumDocuments {
         this.documentMap.set(uriString, document);
     }
 
-    getOrCreateDocument(uri: URI): LangiumDocument {
+    getDocument(uri: URI): LangiumDocument | undefined {
         const uriString = uri.toString();
-        let langiumDoc = this.documentMap.get(uriString);
-        if (langiumDoc) {
-            // The document is already present in our map
-            return langiumDoc;
+        return this.documentMap.get(uriString);
+    }
+
+    async getOrCreateDocument(uri: URI, cancellationToken?: CancellationToken): Promise<LangiumDocument> {
+        let document = this.getDocument(uri);
+        if (document) {
+            return document;
+        }
+        const text = await this.fileSystemProvider.readFile(uri);
```

Comment:
Is there a reason for moving the file content reading from the documentFactory to here?
This feels weird to me, as `update` of `DefaultDocumentFactory` does the FS access, too.